test/build/50_ch-tug.bats

```diff
@@ -234,3 +234,48 @@ EOF
     [[ $output = 'regular file' ]]
     [[ $(cat "${img}/test/source") = 'regular' ]]
 }
+
+@test 'ch-tug image with specified schema' {
+    # Validate that we handle both schema version (1) and (2) image formats.
+    scope standard
+    if ( ! ch-tug --dependencies ); then
+
+        [[ $CH_BUILDER != ch-grow ]]
+        skip "ch-tug missing dependencies"
+    fi
+
+    unpack=$BATS_TMPDIR
+    cache=$unpack/dlcache
+    img=charliecloud%symlink
+
+    export CH_SCHEMA_VER=1
+    img1=$unpack/charliecloud%symlink%v1
+    ch-tug --unpack-dir="$img1" \
+           --dl-cache="$cache" \
+           --no-cache charliecloud/symlink
+    [[ $status -eq 0 ]]
+    cat "${cache}/charliecloud%symlink.manifest.json" | grep -F '"schemaVersion": 1'
+
+    export CH_SCHEMA_VER=2
+    img2=$unpack/charliecloud%symlink%v2
+    ch-tug --unpack-dir="$img2" \
+           --dl-cache="$cache" \
+           --no-cache charliecloud/symlink
+    [[ $status -eq 0 ]]
+    cat "${cache}/charliecloud%symlink.manifest.json" | grep -F '"schemaVersion": 2'
+
+    # The following horror ensures that the same image unpacked via different
+    # schemas are the same.
+    cd "$img1"
+    find "$img" -type f -exec md5sum {} + | sort > "${img1}/contents.txt"
+    cd "$img2"
+    find "$img" -type f -exec md5sum {} + | sort > "${img2}/contents.txt"
+    diff -u "${img1}/contents.txt" "${img2}/contents.txt"
```

Comment:
I think you can just compare two trees directly with `diff`.

Reply:
`diff` complains about broken symlinks and errors with some flavor of`file doesn't exist` error. These symbolic links are pointing to locations that are relative to the container root and are resolvable after `pivot_root`. However, when we probe them in storage they point to places that may not exist.

It wasn't quite clear to tell `diff -r` to treat symbolic links as regular files, i.e., don't follow them.

Reply:
Does `--no-dereference` help?